backend/src/controllers/giftsController/index.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  Query as QueryParam,
+  Request as RequestDecorator,
+  Response as ResponseDecorator
+} from '@decorators/express'
+import { Response } from 'express'
+import GiftService from '../../common/services/gift'
+import AuthMiddleware from '../../middleware/authMiddleware'
+import { IUserEnhancedRequest } from '../../middleware/definitions/authMiddleware'
+import { schemaValidatorFatory } from '../../middleware/schemaValidationMiddleware'
+import { INewGift } from './models'
+import { NewGiftSchema } from './schemas'
+
+@Controller('/gifts', [AuthMiddleware])
+export default class GiftsController {
+  private giftService = new GiftService()
+
+  @Get('/')
```

Comment:
Let's add `schemaValidator` here to validate limit/offset (limit and offset to be ints, limit less than 50)